__tests__/app.test.js

```diff
@@ -50,28 +48,44 @@ describe("GET /api/articles/:article_id", () => {
     expect(body.msg).toBe("bad request");
   });
 });
-describe("GET /api/articles", () => {
-  test("responds with articles a get requested is used", async () => {
-    const { body } = await request(app).get("/api/articles").expect(200);
-    expect(body.articles.length).not.toBe(0);
-    body.articles.forEach((article) => {
-      expect(article).toEqual(
-        expect.objectContaining({
-          article_id: expect.any(Number),
-          title: expect.any(String),
-          topic: expect.any(String),
-          author: expect.any(String),
-          body: expect.any(String),
-          created_at: expect.any(String),
-          votes: expect.any(Number),
-          article_img_url: expect.any(String),
-        })
-      );
+describe("/api/articles", () => {
+  describe("GET /api/articles", () => {
+    test("responds with articles when a get requested is used", async () => {
+      const { body } = await request(app).get("/api/articles").expect(200);
+      expect(body.articles.length).not.toBe(0);
+      body.articles.forEach((article) => {
+        expect(article).toEqual(
+          expect.objectContaining({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+          })
+        );
+      });
+    });
+    test("responds with all articles, in descending order by date", async () => {
+      const { body } = await request(app).get("/api/articles").expect(200);
+      expect(body.articles).toBeSortedBy("created_at", { descending: true });
     });
   });
-  test("responds with all articles, in descending order by date", async () => {
-    const { body } = await request(app).get("/api/articles").expect(200);
-    expect(body.articles).toBeSortedBy("created_at", { descending: true });
+  describe("GET /api/articles (topic query)", () => {
+    test("repond with array of articles filtered by topic", async () => {
+      const { body } = await request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200);
+      expect(body.articles).toBeInstanceOf(Array);
+    });
+    test("should repsond with err when no invalid topic ", async () => {
+      const { body } = await request(app)
+        .get("/api/articles?topic=mit")
+        .expect(404);
+      expect(body.msg).toBe("not found");
+    });
```

Comment:
Good 👍 
We need one more test where we ask for a topic which exists (200) but doesn't have any articles and returns an empty array.